odoo/addons/base/transpiler/transpiler_js.py

```diff
@@ -0,0 +1,393 @@
+        repl = r"\g<space>const \g<identifier> = __exports.\g<identifier> = \g<type> \g<identifier>"
+        if default:
+            pattern = r"^(?P<space>\s*)export\s+default\s+(?P<type>(async\s+)?function|class)\s+(?P<identifier>\w+)"
+            repl = r"""\g<space>const \g<identifier> = __exports[Symbol.for("default")] = \g<type> \g<identifier>"""
+        p = re.compile(pattern, re.MULTILINE)
+        self.content = p.sub(repl, self.content)
+
+    def convert_variable_export(self, default=False):
+        """
+        Transpile variables that are being exported.
+
+        before: export let name =
+        after: let name = __exports.name =
+        (same with var and const)
+
+        if default:
+        before: export let name =
+        after: let name = __exports[Symbol.for("default")] =
+
+        :param default: should we convert the default exports as well ?
+        """
+        pattern = r"^(?P<space>\s*)export\s+(?P<type>let|const|var)\s+(?P<identifier>\w+)\s*="
+        repl = r"\g<space>\g<type> \g<identifier> = __exports.\g<identifier> ="
+        if default:
+            pattern = r"^(?P<space>\s*)export\s+default\s+(?P<type>let|const|var)\s+(?P<identifier>\w+)\s*="
+            repl = r"""\g<space>\g<type> \g<identifier> = __exports[Symbol.for("default")] ="""
+        p = re.compile(pattern, re.MULTILINE)
+        self.content = p.sub(repl, self.content)
+
+    def convert_list_export(self):
+        """
+        Transpile exports of multiple elements
+
+        before: export { a, b, c as x }
+        after: Object.assign(__exports, { a, b, x: c })
+        """
+        p = re.compile(r"^(?P<space>\s*)export\s*(?P<list>{(\s*\w+\s*,?\s*)*}\s*)", re.MULTILINE)
+
+        def repl(matchobj):
+            d = matchobj.groupdict()
+            list_process = "{" + ", ".join([convert_as(val) for val in d.get("list")[1:-1].split(",")]) + "}"
+            space = d.get("space", "")
+            return f"{space}Object.assign(__exports, {list_process})"
+        self.content = p.sub(repl, self.content)
+
+    def convert_from_export(self):
+        """
+        Transpile exports coming from another source
+
+        before: export { a, b, c as x } from "some/path.js"
+        after: { a, b, c } = {require("some/path.js"); Object.assign(__exports, { a, b, x: c });}
+        """
+        p = re.compile(r"^(?P<space>\s*)export\s*(?P<list>{(\s*\w+\s*,?\s*)*})\s*from\s*(?P<path>(\".*\")|('.*')|(`.*`))", re.MULTILINE)
+
+        def repl(matchobj):
+            d = matchobj.groupdict()
+            list_clean = "{" + ", ".join([remove_as(val) for val in d.get("list")[1:-1].split(",")]) + "}"
+            list_process = "{" + ", ".join([convert_as(val) for val in d.get("list")[1:-1].split(",")]) + "}"
+            space = d.get("space", "")
+            path = d.get("path")
+            return f"{space}" + "{" + f"const {list_clean} = require({path});Object.assign(__exports, {list_process})" + ";}"
```

Comment:
Concatenations seem a bit redundant here, can just double up the literal braces to escape them (or use old-style formatting).